### DISCUSSION TITLE
Can the MS2130 not capture audio?

### DISCUSSION THREAD

**Discussion by rorosaurus** [awawa-dev/HyperHDR#866] [Q&A]
My HyperHDR setup has been working great! I've only recently started to play with some of the more advanced features though, and I realize I'm not getting any audio reactivity from the sound capture devices available to me in the dropdown on the effects page: 

![image](https://github.com/awawa-dev/HyperHDR/assets/908662/3151d200-c54b-40d5-bf08-1b823a327870)

I've gone to the remote page, and changed to a "music effect" and tested each of these sound devices with no response from my LEDs each time. Am I missing something obvious?

My HDMI capture is the MS2130 model B linked in the discussions on here, plugged into HyperHDR running on a RPi 4. The LEDs are controlled over the network with WLED, and they work fine with HDMI video input capture for the normal ambilight type effect!

**Comment by awawa-dev** [comment]
https://github.com/awawa-dev/HyperHDR/discussions/499#discussioncomment-5076759
https://github.com/awawa-dev/HyperHDR/wiki/Quick-start#configuring-and-enabling-sound-visualization-effects

**Reply by rorosaurus** [reply]
Thank you! That's interesting, I'm using USB3.0. The equalizer is flat (non existent) but there's also no log about silence. There is a frame error though.

~~~~
2024-05-25T17:37:41.960Z [V4L2:USB3 VIDEO: US] Frame too small: 0 != 691200
~~~~

Full log: https://pastebin.com/E3bNGZ00

**Reply by awawa-dev** [reply]
Hi
Seems everything is OK in logs (that error message is irrelevant when the grabber is starting)
```
2024-05-25T17:30:19.256Z [SOUND_GRABBER] Sound device is starting
2024-05-25T17:30:19.256Z [SOUND_GRABBER] Opening device: plughw:CARD=VideoDEV=0 | USB3 Video USB AudioHardware device with all software conversions | Input
2024-05-25T17:30:19.256Z [SOUND_GRABBER] Sound device has started (handle: 10)
2024-05-25T17:30:19.256Z [EFFECT0(Music:...)] Begin playing the music effect with priority: 1
2024-05-25T17:30:19.258Z [SOUND_GRABBER] Sound period size = 1024
2024-05-25T17:30:19.258Z [SOUND_GRABBER] Sound buffer size = 2048
2024-05-25T17:30:19.311Z [SOUND_GRABBER] Audio Stream: audio data successfully captured and sound detected.
```
Which version are you using? I'll test it again.

Besides, you have a problem with the power supply and the power limiter in wled is turned on (then wled renders the LEDs in its own way and it is not 1:1 with what we sent, e.g. HyperHDR anti-flickering filter may not work properly)

**Reply by rorosaurus** [reply]
Okay, after a bit more experimenting, I've identified my HDMI splitter/scaler as the issue. When I bypass it directly, HyperHDR gets the audio data.

What's strange is, I can verify the scaled output still has the audio by swapping my TV's HDMI cable to use the splitter scaled output. I confirmed it's not the HDMI cables. But this seems to suggest the splitter garbles the audio in a way the TV can still use it, but HyperHDR gets tripped up.

For future readers, my EZCOO splitter, which I can't recommend due to this issue and not supporting CEC: https://a.co/d/1k15bgN

**Reply by awawa-dev** [reply]
Maybe it's a protocol issue? I doubt the splitter can do Dolby Atmos or DTS to PCM for just one secondary output. ms2130 probably requires raw PCM but I'm not sure about it.
